### PR TITLE
Implement `getChainTip` Ogmios query.

### LINKED ISSUE #169: Add Ogmios' `chainTip` to `QueryM` (to replicate `currentSlot` function)
As part of the script lookups PR https://github.com/Plutonomicon/cardano-browser-tx/pull/148, we replicated most of the `TimeSlot` module with the exception of `currentSlot` https://playground.plutus.iohkdev.io/doc/haddock/plutus-ledger/html/Ledger-TimeSlot.html#v:currentSlot. There is a placeholder for this in `Types.Interval` `currentSlot :: SlotConfig -> Effect Slot`.

### PR DESCRIPTION
**Pull request by adamczykm**
Closes #169

### CODE REVIEW

File: src/Aeson.purs
Comment (ngua) [inline src/Aeson.purs:348]: I wonder if there's any way we can factor out some of these instances for integral types, like `(String -> Maybe a) -> Aeson -> Either JsonDecodeError a`

File: test/AffInterface.purs
Comment (ngua) [inline test/AffInterface.purs:12]: Could you revert the formatting change?

File: src/Types/JsonWsp.purs
Comment (ngua) [inline src/Types/JsonWsp.purs:315]: Is there a reason to use `Untagged.Union` instead of a sum type?

File: src/Aeson.purs
Comment (ngua) [inline src/Aeson.purs:392]: Why not use `<|>`?

Comment (adamczykm) [reply src/Aeson.purs:392]: I did start with (<|>) but then had some typechecking issues and forgot to fix it back.

File: src/Types/JsonWsp.purs
Comment (ngua) [reply src/Types/JsonWsp.purs:315]: I thought it might be easier for users to define this as a sum type, but either way works I suppose

Comment (adamczykm) [reply src/Types/JsonWsp.purs:315]: In general, I'm in favour of using Variant for such purposes, but it is a little bit more vocal. I will change it to a sum type.

File: src/Aeson.purs
Comment (klntsky) [inline src/Aeson.purs:386]: [InOneOf](https://pursuit.purescript.org/packages/purescript-untagged-union/0.3.0/docs/Untagged.Union#t:InOneOf) has an instance of form `InOneOf b a b` for any types `a` and `b`, so this seem to be a useless constraint here. Am I right?

**Comment by ngua** [comment]
We have an issue with CI. I opened a PR to fix it, so we can merge this after merging in those changes

**Review by ngua — APPROVED** [review]

**Pull request merged**
